ruby/src/main/java/com/google/protobuf/jruby/RubyRepeatedField.java

```diff
@@ -268,17 +268,12 @@ public IRubyObject eq(ThreadContext context, IRubyObject other) {
     @JRubyMethod
     public IRubyObject each(ThreadContext context, Block block) {
         this.storage.each(context, block);
-        return context.runtime.getNil();
+        return this.toArray(context);
     }
 
+
     @JRubyMethod(name = {"to_ary", "to_a"})
     public IRubyObject toArray(ThreadContext context) {
-        for (int i = 0; i < this.storage.size(); i++) {
```

Comment:
Good catch on removing this -- `nil` shouldn't be in the array to begin with. Thanks!